### PR TITLE
feat: notify miner to skip obviously large txns without executing them twice

### PR DESCRIPTION
**Pull request by omerfirmak**
## 1. Purpose or design rationale of this PR

miner will skip a txn if it counter-based CCC says so, but there might be cases where counter-based CCC thinks a txn is OK but Rust CCC thinks otherwise.

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [ ] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [ ] This PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: rollup/ccc/checker.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func (ccc *Checker) ApplyBlock(traces *types.BlockTrace) (*types.RowConsumption,
 		return nil, ErrUnknown
 	}
 	if !result.AccRowUsage.IsOk {
-		return nil, ErrBlockRowConsumptionOverflow
+		return (*types.RowConsumption)(&result.AccRowUsage.RowUsageDetails), ErrBlockRowConsumptionOverflow
```
Comment (0xmountaintop) [inline rollup/ccc/checker.go:156]: similar to here, should we also update L116?

**Comment by 0xmountaintop** [comment]
so basically the workflow in this scenario is

counter-based ccc -> rust ccc -> shouldSkip -> reorg and add logic to directly skip this tx


?

**Comment by omerfirmak** [comment]
> so basically the workflow in this scenario is
> 
> counter-based ccc -> rust ccc -> shouldSkip -> reorg and add logic to directly skip this tx
> 
> ?

Yes, that is pretty much it. Should be fairly reliable in terms of catching skipped txns without causing too many reorgs.

File: rollup/ccc/checker.go (hunk 1 above)
Comment (omerfirmak) [reply rollup/ccc/checker.go:156]: I actually meant to change L116 😮‍💨  Thanks!

**Review by 0xmountaintop — APPROVED** [review]

File: rollup/ccc/async_checker.go [hunk 2]
```diff
@@ -157,27 +158,32 @@ func (c *AsyncChecker) checkerTask(block *types.Block, ccc *Checker, forkCtx con
...
-		rc, err = c.checkTxAndApply(parent, header, statedb, gasPool, tx, ccc)
+		var curRc *types.RowConsumption
+		curRc, err = c.checkTxAndApply(parent, header, statedb, gasPool, tx, ccc)
 		if err != nil {
 			err = &ErrorWithTxnIdx{
 				txIdx: uint(txIdx),
 				err:   err,
+				// if the txn is the first in block or the additional resource utilization caused
+				// by this txn alone is enough to overflow the circuit, skip
+				shouldSkip: txIdx == 0 || curRc.Difference(*accRc).IsOverflown(),
```
Comment (Thegaram) [inline rollup/ccc/async_checker.go:175]: Does this make the `prioritizedTx` feature on the worker useless?

**Pull request merged**